src/renderer/dx/DxRenderer.cpp

```diff
@@ -1786,6 +1786,14 @@ CATCH_RETURN();
     return Viewport::FromDimensions(viewInPixels.Origin(), { widthInChars, heightInChars });
 }
 
+[[nodiscard]] Viewport DxEngine::GetViewportInPixels(const Viewport& viewInCharacters) noexcept
+{
+    const short widthInPixels = gsl::narrow_cast<short>(viewInCharacters.Width() * _glyphCell.width());
+    const short heightInPixels = gsl::narrow_cast<short>(viewInCharacters.Height() * _glyphCell.height());
+
+    return Viewport::FromDimensions(viewInCharacters.Origin(), { widthInPixels, heightInPixels });

```

Comment:
I feel like an overflow of `short` is not out of the question with this multiplication and that this method should either throw in that circumstance, or the consumer should call both `GetViewportInChars` and `GetFontSize` and handle the multiplication on its own side, resolving the overflow however it chooses (though hopefully not letting it happen silently with `narrow_cast`.)

Reply:
Is that really necessary? `short` has an upper limit of 32,000 and considering these are pixels that would mean that someone is attempting to calculate the size of a terminal that fits a 32k-wide screen

Reply:
Four 8k displays would nearly reach the limit. Give it a few years and I bet they come up with an even higher resolution. Don't forget 640KB of memory should have been enough for everybody.

All I'm really saying is that using `base::saturated_cast<short>` on the result to lock it to `SHORT_MAX` when it's too big is probably better than letting it ever go negative or to some weird value when it one day overflows by completely ignoring the overflow through `gsl::narrow_cast<short>`.

Reply:
You can put that in this helper instead of the initial proposals with the throws.